mantle/kola/tests/coretest/core.go

```diff
@@ -92,6 +134,41 @@ func init() {
 	})
 }
 
+func uefiWithBasicTests(c cluster.TestCluster) {
+	runBasicTests(c, uefi, false)
+}
+
+func uefiSecureWithBasicTests(c cluster.TestCluster) {
+	runBasicTests(c, uefiSecure, false)
+}
+
+func nvmeBasicTests(c cluster.TestCluster) {
+	runBasicTests(c, bios, true)
+}
+
+func runBasicTests(c cluster.TestCluster, firmware string, nvme bool) {
+	var err error
+
+	options := platform.QemuMachineOptions{
+		Firmware: firmware,
+		Nvme:     nvme,
+	}
+	switch pc := c.Cluster.(type) {
+	// These cases have to be separated because when put together to the same case statement
+	// the golang compiler no longer checks that the individual types in the case have the
+	// NewMachineWithQemuOptions function, but rather whether platform.Cluster
+	// does which fails
+	case *qemu.Cluster:
+		_, err = pc.NewMachineWithQemuOptions(nil, options)
+	default:
+		panic("Unsupported cluster type")
+	}
+	if err != nil {
+		c.Fatal(err)
+	}
```

Comment:
I think you'll need to leep the Machine return value from `NewMachineWithQemuOptions()`, and then around here, call `scpKolet()` to copy kolet into the machine (I guess you might have to make that function public).

Reply:
Thank you, that makes sense, I should have looked more closely at what the harness was doing to the cluster machines.

Reply:
100%